main/main.cpp

```diff
@@ -1881,9 +1880,11 @@ bool Main::start() {
 				script = args[i + 1];
 #ifdef TOOLS_ENABLED
 			} else if (args[i] == "--doctool") {
-				doc_tool = args[i + 1];
-				for (int j = i + 2; j < args.size(); j++) {
-					removal_docs.push_back(args[j]);
+				doc_tool_path = args[i + 1];
+				if (doc_tool_path.begins_with("-")) {
```

Comment:
For the reference, this is something that could be done further up after `else if (i < (args.size() - 1)) {` for all "paired" parameters, but it's more risky so I'm keeping it localized.

The proper fix anyway is to scrap all this and redo the argument parsing with a proper parser (#44594).